core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -82,7 +90,6 @@ public TableDataMap createAndRegisterDataMap(AbsoluteTableIdentifier identifier,
     List<TableDataMap> tableDataMaps = allDataMaps.get(table);
     if (tableDataMaps == null) {
       tableDataMaps = new ArrayList<>();
-      allDataMaps.put(table, tableDataMaps);
```

Comment:
FYI: 

should move this line to the end, otherwise in concurrent scenario, the other thread will get empty (not null) `tableMaps` in `getDataMap`, which will result in `Datamap does not exist` exception.